File: tarea7/java/arbol/ArbolBinario.java
package arbol;

public class ArbolBinario{
  Nodo raiz;

  public ArbolBinario(){
    raiz = null;
  }

  public ArbolBinario(Nodo raiz){
    this.raiz = raiz;
  }

  Nodo getRaiz(){
    return raiz;
  }

  boolean isEmpty(){
    return raiz == null;
  }

  Nodo nuevoArbol(Nodo hijo_izq, Object valor, Nodo hijo_der){
    return new Nodo(hijo_izq, valor, hijo_der);
  }
}

File: tarea7/ruby/ArbolBinario.rb
#module arbol
#require_relative 'Nodo'

require 'java'
java_import 'ruby.Nodo'

  class ArbolBinario
    @raiz

    def initialize(raiz = nil)
      @raiz = raiz
    end

    def raiz
      @name
    end

    def raiz=(val)
      @name = val
    end

    def is_empty
      @raiz == nil
    end

    def nuevo_arbol(hijo_izq, valor, hijo_der)
      Nodo.new(hijo_izq, valor, hijo_der)
    end
  end
#end
